packages/frontend/editor-ui/src/features/workflows/canvas/composables/useCanvasMapping.ts

```diff
@@ -401,7 +401,7 @@ export function useCanvasMapping({
 				return acc;
 			}, {});
 		},
-		{ throttle: CANVAS_EXECUTION_DATA_THROTTLE_DURATION, immediate: true },
+		{ throttle: CANVAS_EXECUTION_DATA_THROTTLE_DURATION, immediate: true, deep: true },
```

Comment:
Since execution data can be large, deep watch may cause a performance issue. It may worthwhile to test this solution with some of workflows that we used for the perf improvement work 🤔